### DISCUSSION TITLE
When referring to notes in prompts, it's impossible to distinguish between notes with the same name.

### DISCUSSION THREAD

**Discussion by Wooden-Gear** [logancyang/obsidian-copilot#1170] [Q&A]
在提示词中引用笔记，无法区分同名笔记。
我在提示词中引用某个笔记的时候，所有同名的笔记都会被一起引用。
是不是可以自动识别当前的活动笔记，然后只引用跟活动笔记同一目录的那个笔记，而不引用其他目录的同名笔记。

---

When referencing notes in the prompt, it's impossible to distinguish between notes with the same name.
When I reference a note in the prompt, all notes with the same name are referenced together.
Can it automatically identify the current active note and only reference the note in the same directory as the active note, instead of referencing other notes with the same name in different directories?

**Comment by logancyang** [comment]
hmm.. It's not a good practice to have notes with the same name. The request also doesn't make sense to me, why the note in the same directory? why not same tag? why not any random criteria?

**Comment by Wooden-Gear** [comment]
> hmm.. It's not a good practice to have notes with the same name. The request also doesn't make sense to me, why the note in the same directory? why not same tag? why not any random criteria?

![image](https://github.com/user-attachments/assets/37a47f28-9419-4de5-9fba-deeec6095b0b)

箭头表示提示词对文件的调用，图片上还是少了。
如果两个项目的文件名不统一的话，同样的提示词调用不同的文件，针对不同的项目，排列组合下来，得创建多少条啊！

Arrows indicate the prompt calling the file, but there are still some missing on the image.
If the file names of two projects are not consistent, the same prompt will call different files for different projects. The number of combinations that need to be created would be huge!

**Comment by logancyang** [comment]
Ok, so you want the prompt to reference different filenames as variables while keeping the instructions the same.

That's ok, you can just do something like this

```
You are tasked with producing a prompt with provided filenames below. Fill in the file name at `<>` in order and return it:

PROMPT: do action 1 to <> according to <> and then do action 2 according to <>
FILENAMES: {filename_prefix1}, {filename_prefix2}, {filename_prefix3}
```

Call it with `/` to add to your chat input. Then you can tweak the filenames a little bit and use the resulting prompt to do what you want.
